### PR TITLE
Force a rekey on join if stream and table partitions dont match

### PR DESCRIPTION
**Pull request by rodesai**

### CODE REVIEW

File: ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java [hunk 1]
```diff
@@ -52,6 +56,16 @@ public int getNumPartitions() {
     public short getReplicatonFactor() {
       return replicatonFactor;
     }
+
+    public TopicDescription getDescription() {
+      List<TopicPartitionInfo> partitionInfoList = new LinkedList<>();
```
Comment (big-andy-coates) [inline ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java]: Why `LinkedList`?

File: ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java [hunk 2]
```diff
@@ -52,6 +56,16 @@ public int getNumPartitions() {
...
+      Node node = new Node(0, "localhost", 9091);
+      for (int i = 0; i < numPartitions; i++) {
+        partitionInfoList.add(
+            new TopicPartitionInfo(0, node, new LinkedList<>(), new LinkedList<>()));
```
Comment (big-andy-coates) [inline ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java]: Can these `new LinkedList`s just be `Collections.emptyList()`, or do they need to be mutable?

Comment (rodesai) [reply ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java]: Nope. Still learning the latest and greatest java idioms :)

File: ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java (hunk 1 above)
Comment (rodesai) [reply ksql-engine/src/test/java/io/confluent/ksql/util/FakeKafkaTopicClient.java]: No special reason. Is there a more idiomatic way to construct a implementation-insensitive mutable list?

**Pull request merged**